plugins/processors/prometheusadapter/processor.go

```diff
@@ -5,24 +5,194 @@ package prometheusadapter
 
 import (
 	"context"
+	"os"
+	"strings"
 
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
 	"go.uber.org/zap"
+
+	"github.com/aws/amazon-cloudwatch-agent/internal/util/collections"
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/prometheusadapter/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+	"github.com/prometheus/common/model"
 )
 
 type prometheusAdapterProcessor struct {
 	*Config
-	logger *zap.Logger
+	logger          *zap.Logger
+	deltaCalculator *internal.DeltaCalculator
 }
 
 func newPrometheusAdapterProcessor(config *Config, logger *zap.Logger) *prometheusAdapterProcessor {
 	d := &prometheusAdapterProcessor{
-		Config: config,
-		logger: logger,
+		Config:          config,
+		logger:          logger,
+		deltaCalculator: internal.NewDeltaCalculator(),
 	}
 	return d
 }
 
 func (d *prometheusAdapterProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+
+	d.preprocessFilter(md)
+
+	md.ResourceMetrics().RemoveIf(func(rm pmetric.ResourceMetrics) bool {
+		rma := rm.Resource().Attributes()
+		rm.ScopeMetrics().RemoveIf(func(sm pmetric.ScopeMetrics) bool {
+			sm.Metrics().RemoveIf(func(m pmetric.Metric) bool {
+				return d.processMetric(m, rma)
+			})
+			return sm.Metrics().Len() == 0
+		})
+		return rm.ScopeMetrics().Len() == 0
+	})
+
+	d.postprocessFilter(md)
+
 	return md, nil
 }
+
+func (d *prometheusAdapterProcessor) preprocessFilter(md pmetric.Metrics) {
+
+	md.ResourceMetrics().RemoveIf(func(rm pmetric.ResourceMetrics) bool {
+		rm.ScopeMetrics().RemoveIf(func(sm pmetric.ScopeMetrics) bool {
+			sm.Metrics().RemoveIf(func(m pmetric.Metric) bool {
+
+				const (
+					keep = false
+					drop = true
+				)
+
+				// for backwards compatibility with legacy Telegraf receiver, we want to drop some extraneous metrics
+				extraneousMetrics := collections.Set[string]{
+					"scrape_duration_seconds":               {},
+					"scrape_samples_post_metric_relabeling": {},
+					"scrape_samples_scraped":                {},
+					"scrape_series_added":                   {},
+					"up":                                    {},
+				}
```

Comment:
I meant as a package level variable. This is a static set, so we shouldn't define and allocate a new map on each loop.

```suggestion
	var scrapeStatMetrics = collections.NewSet("scrape_duration_seconds", "scrape_samples_post_metric_relabeling", "scrape_samples_scraped", "scrape_series_added", "up")
```

Reply:
Won't compiler make that optimization for us and not actually allocate a new map each loop?